yarn/src/main/scala/org/apache/spark/deploy/yarn/security/AMDelegationTokenRenewer.scala

```diff
@@ -171,10 +174,10 @@ private[yarn] class AMDelegationTokenRenewer(
     keytabLoggedInUGI.doAs(new PrivilegedExceptionAction[Void] {
       // Get a copy of the credentials
       override def run(): Void = {
-        val nns = YarnSparkHadoopUtil.get.getNameNodesToAccess(sparkConf) + dst
-        hadoopUtil.obtainTokensForNamenodes(nns, freshHadoopConf, tempCreds)
-        hadoopUtil.obtainTokenForHiveMetastore(sparkConf, freshHadoopConf, tempCreds)
-        hadoopUtil.obtainTokenForHBase(sparkConf, freshHadoopConf, tempCreds)
+        val nearestNextTime = credentialManager.obtainCredentials(freshHadoopConf, tempCreds)
+        require(nearestNextTime > System.currentTimeMillis(),
+          s"Time of next renewal $nearestNextTime is earlier than now")
+        timeOfNextRenewal = nearestNextTime
```

Comment:
driverTokenRenewerRunnable is hardcoded to renew every hour: 
              delegationTokenRenewer.schedule(this, 1, TimeUnit.HOURS)

We should probably make this configurable.  Ideally nothing needs to be renewed that often but if we are wanting to support arbitrary services making it configurable would make sense.